### DISCUSSION TITLE
ec2 AMI/종료방지설정

### DISCUSSION THREAD

**Discussion by lina1919** [SWM-Space-Odyssey/WeddingMate_BackEnd#117] [General]
## 문제 상황
- ec2 인스턴스를.. 실수로.. 종료시킴^^
- code deploy 를 적용시키는 상황에서 auto scaling된 인스턴스를 중지 시킬 일이 많았다. 그래서 여러 개의 인스턴스를 한번에 종료시키는 상황에서, 실수롤 개발서버 인스턴스를 종료시킴.
	*- 종료 vs 중지* 인간적으로 말이 너무 헷갈린거아닙니까
- 종료 시킨 인스턴스 다시 실행할 수 없을까?
	- NO....😭
https://repost.aws/ko/knowledge-center/recovery-terminated-instance
> 원본 Amazon EC2 인스턴스 또는 종료 프로세스의 일부로 삭제된 볼륨은 복구할 수 없습니다.
### 인스턴스 AMI
- 인스턴스 ami를 설정해두어 빠르게 복구할 수 있었다.
	- 종료된 Amazon EC2 인스턴스에서 생성된 AMI 백업을 사용하여 [대체 EC2 인스턴스를 시작](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-launch-snapshot.html)
#### AMI란?
일종의 EC2를 시작하기 위한 기본 세팅의 모음이다.
- EC2 인스턴스를 실행하기 위해 필요한 정보를 모은 단위
    - OS, 아키텍쳐 타입, 저장공간 용량 등

#### AMI 만드는 과정
- EBS를 스냅샷을 뜸(OS 파일 시작권한이 포함) 그 후 S3에 저장한다.(내부적으로 자동진행)
- 그 후 AMI 등록
- AMI는 기본적으로 제공되는것도 있고 내가 원하는 것에 복사해 나만의 AMI 제작이 가능, 다른사람한테 공유가 가능하다.
- 백업용으로 최적의 구성을 만든 후 AMI에 저장할 수 있다.
- AMI 가지고 새로운 EC2 실행이 가능하고 AMI 복사 후 다른사람에게 넘겨주는게 가능

#### 스냅샷이란
스냅샷은 incremental 이다. 100GB 짜리 EBS에 대한 이미지를 생성  
그 후 바뀔때마다 기존이미지에서 바뀐 부분만 저장(생성,삭제 모두 다)하기 때문에 용량이 비약적으로 줄어든다. 돈을 아낄 수 있다.

### EC2 종료 방지 설정
- 그런 일 방지를 위해 인스턴스 종료 방지 기능 활성화!
	
![](https://i.imgur.com/dwcNGXx.png)

## 마치면서
- 내 눈, 내 손 그 어떤 것도 믿지말자.
- 무조건 백업하자.

**Comment by nitronium102** [comment]
눈물없이 볼 수 없는 인프라.....고생했어👍👍👍
